server/src/main/java/org/apache/cassandra/sidecar/utils/SidecarClientProvider.java

```diff
@@ -76,6 +77,21 @@ public SidecarClient get()
         return client;
     }
 
+    public void close(Promise<Void> completion)
+    {
+        LOGGER.info("Closing Sidecar Client...");
+        try
+        {
+            client.close();
+            Thread.sleep(100);
```

Comment:
This (sleep) is interesting. Why?

Reply:
@frankgh identified a bug in vertx itself, that completing promises so close together may cause problems.